Panic when `secret` is not valid string

**Describe the bug**
When running `okteto build` with an invalid format for the secret string, a panic is returned

Found while testing some scenarios on okteto build command:

```
running: okteto build --progress plain --secret secret
 i  Using teresa @ - as context
 i  Building 'Dockerfile' in -:443...
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/okteto/okteto/pkg/cmd/build.parseTempSecrets({0x140015bfad0, 0x23}, 0x14001048c80)
	github.com/okteto/okteto/pkg/cmd/build/build.go:418 +0x47c
github.com/okteto/okteto/pkg/cmd/build.(*OktetoBuilder).buildWithOkteto(0x4d?, {0x103756b60, 0x1050c1320}, 0x14001048c80)
	github.com/okteto/okteto/pkg/cmd/build/build.go:132 +0x3dc
github.com/okteto/okteto/pkg/cmd/build.(*OktetoBuilder).Run(0x1400154c1a0?, {0x103756b60, 0x1050c1320}, 0x14001048c80)
	github.com/okteto/okteto/pkg/cmd/build/build.go:66 +0x150
github.com/okteto/okteto/cmd/build/v1.(*OktetoBuilder).Build(0x14001688000, {0x103756b60, 0x1050c1320}, 0x14001048c80)
	github.com/okteto/okteto/cmd/build/v1/build.go:104 +0x320
github.com/okteto/okteto/cmd/build/v2.(*OktetoBuilder).buildSvcFromDockerfile(0x14001684000, {0x103756b60, 0x1050c1320}, 0x14001570000, {0x14001552b85, 0xb}, 0x140004b83c0)
	github.com/okteto/okteto/cmd/build/v2/build.go:249 +0x274
github.com/okteto/okteto/cmd/build/v2.(*OktetoBuilder).buildService(0x103743c08?, {0x103756b60, 0x1050c1320}, 0x14001570000, {0x14001552b85, 0xb}, 0x10371ba00?)
	github.com/okteto/okteto/cmd/build/v2/build.go:224 +0xa8
github.com/okteto/okteto/cmd/build/v2.(*OktetoBuilder).Build(0x14001684000, {0x103756b60, 0x1050c1320}, 0x140004b83c0)
	github.com/okteto/okteto/cmd/build/v2/build.go:184 +0x888
github.com/okteto/okteto/cmd/build.Build.func1(0x140013eac00?, {0x140013b5800?, 0x4?, 0x10262377e?})
	github.com/okteto/okteto/cmd/build/command.go:105 +0x2cc
github.com/spf13/cobra.(*Command).execute(0x14000248780, {0x140013b57c0, 0x4, 0x4})
	github.com/spf13/cobra@v1.5.0/command.go:872 +0x560
github.com/spf13/cobra.(*Command).ExecuteC(0x1400032e500)
	github.com/spf13/cobra@v1.5.0/command.go:990 +0x318
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	github.com/okteto/okteto/main.go:156 +0x9e8

```


**To Reproduce**
Steps to reproduce the behavior:

1. Go to any repository, for example `go-getting-started`
1. Use the build command with the `--secret` flag and use a format not compliant
1. A panic is returned

**Expected behavior**
A clean message of error explaining the format of the secret input

**Additional context**

Found the line throwing the panic https://github.com/okteto/okteto/blob/master/pkg/cmd/build/build.go#L419 
We should check if the Split done previously has indeed valid output and not `nil`

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -44,6 +44,13 @@ const (
 	doubleDockerfileWarning string = "Build '%s': Two Dockerfiles discovered in both the root and context path, defaulting to '%s/%s'"
 )
 
+var (
+	errDockerDaemonConnection = oktetoErrors.UserError{
+		E:    fmt.Errorf("cannot connect to Docker Daemon"),
+		Hint: "Please start the Docker Daemon or configure a builder endpoint with 'okteto context --builder BUILDKIT_URL",
+	}
+)
+
 // OktetoBuilderInterface runs the build of an image
 type OktetoBuilderInterface interface {
 	Run(ctx context.Context, buildOptions *types.BuildOptions) error
@@ -235,10 +242,7 @@ func translateDockerErr(err error) error {
 		return nil
 	}
 	if strings.HasPrefix(err.Error(), "failed to dial gRPC: cannot connect to the Docker daemon") {
-		return oktetoErrors.UserError{
-			E:    fmt.Errorf("cannot connect to Docker Daemon"),
-			Hint: "Please start the Docker Daemon or configure a builder endpoint with 'okteto context --builder BUILDKIT_URL",
-		}
+		return errDockerDaemonConnection
 	}
 	return err
 }

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -16,6 +16,7 @@ package build
 import (
 	"bufio"
 	"context"
+	"encoding/csv"
 	"fmt"
 	"os"
 	"path/filepath"
@@ -35,6 +36,7 @@ import (
 	"github.com/okteto/okteto/pkg/registry"
 	"github.com/okteto/okteto/pkg/types"
 	"github.com/pkg/errors"
+	"github.com/spf13/afero"
 )
 
 const (
@@ -130,7 +132,7 @@ func (ob *OktetoBuilder) buildWithOkteto(ctx context.Context, buildOptions *type
 	defer os.RemoveAll(secretTempFolder)
 
 	// inject secrets to buildkit from temp folder
-	if err := parseTempSecrets(secretTempFolder, buildOptions); err != nil {
+	if err := replaceSecretsSourceEnvWithTempFile(afero.NewOsFs(), secretTempFolder, buildOptions); err != nil {
 		return err
 	}
 
@@ -405,83 +407,76 @@ func GetVolumesToInclude(volumesToInclude []model.StackVolume) []model.StackVolu
 	return result
 }
 
-// parseSecretFromString returns id and src from the given secret string
-// format should be id=<name>,src=<source file>
-// if no id or src if found, error is returned
-func parseSecretFromString(s string) (string, string, error) {
-	_, afterId, found := strings.Cut(s, "id=")
-	if !found {
-		return "", "", fmt.Errorf("secret should have an id")
-	}
-	id, _, _ := strings.Cut(afterId, ",")
-	if id == "" {
-		return "", "", fmt.Errorf("secret id can not be empty")
-	}
 
-	_, afterSrc, found := strings.Cut(s, "src=")
-	if !found {
-		return "", "", fmt.Errorf("secret should have a source")
-	}
-	srcFileName, _, _ := strings.Cut(afterSrc, ",")
-	if srcFileName == "" {
-		return "", "", fmt.Errorf("secret source should not be empty")
-	}
 
-	return id, srcFileName, nil
-}
-
-// parseTempSecrets reads the content of the src of a secret and replaces the envs to mount into dockerfile
-func parseTempSecrets(secretTempFolder string, buildOptions *types.BuildOptions) error {
+// replaceSecretsSourceEnvWithTempFile reads the content of the src of a secret and replaces the envs to mount into dockerfile
+func replaceSecretsSourceEnvWithTempFile(fs afero.Fs, secretTempFolder string, buildOptions *types.BuildOptions) error {
 	// for each secret at buildOptions extract the src
 	// read the content of the file
-	// create a new file under tempFolder with the expanded content
+	// create a new file under secretTempFolder with the expanded content
 	// replace the src of the secret with the tempSrc
 	for indx, s := range buildOptions.Secrets {
-		secretId, srcFileName, err := parseSecretFromString(s)
-		if err != nil {
-			return err
-		}
-		// read source file
-		srcFile, err := os.Open(srcFileName)
+		csvReader := csv.NewReader(strings.NewReader(s))
+		fields, err := csvReader.Read()
 		if err != nil {
 			return err
 		}
 
-		// create temp file
-		tmpfile, err := os.CreateTemp(secretTempFolder, "secret-")
-		if err != nil {
-			return err
-		}
-		writer := bufio.NewWriter(tmpfile)
-
-		sc := bufio.NewScanner(srcFile)
-		for sc.Scan() {
-			// expand content
-			srcContent, err := model.ExpandEnv(sc.Text(), true)
-			if err != nil {
-				return err
+		newFields := make([]string, len(fields))
+		for indx, field := range fields {
+			key, value, found := strings.Cut(field, "=")
+			if !found {
+				return fmt.Errorf("format error")
 			}
 
-			// save expanded to temp file
-			_, err = writer.Write([]byte(fmt.Sprintf("%s\n", srcContent)))
-			if err != nil {
-				return fmt.Errorf("unable to write to temp file: %s", err)
+			if key == "src" || key == "source" {
+				tempFileName, err := createTempFileWithExpandedEnvsAtSource(fs, value, secretTempFolder)
+				if err != nil {
+					return err
+				}
+				value = tempFileName
 			}
-			writer.Flush()
+			newFields[indx] = fmt.Sprintf("%s=%s", key, value)
 		}
-		if err := tmpfile.Close(); err != nil {
-			return err
-		}
-		if err := srcFile.Close(); err != nil {
-			return err
-		}
-		if sc.Err() != nil {
-			return sc.Err()
-		}
+		buildOptions.Secrets[indx] = strings.Join(newFields, ",")
+	}
+	return nil
+}
 
-		// replace src
-		buildOptions.Secrets[indx] = fmt.Sprintf("id=%s,src=%s", secretId, tmpfile.Name())
+// createTempFileWithExpandedEnvsAtSource creates a temp file with the expanded values of envs in local secrets
+func createTempFileWithExpandedEnvsAtSource(fs afero.Fs, sourceFile, tempFolder string) (string, error) {
+	srcFile, err := fs.Open(sourceFile)
+	if err != nil {
+		return "", err
 	}
 
-	return nil
+	// create temp file
+	tmpfile, err := afero.TempFile(fs, tempFolder, "secret-")
+	if err != nil {
+		return "", err
+	}
+
+	writer := bufio.NewWriter(tmpfile)
+
+	sc := bufio.NewScanner(srcFile)
+	for sc.Scan() {
+		// expand content
+		srcContent, err := model.ExpandEnv(sc.Text(), true)
+		if err != nil {
+			return "", err
+		}
+
+		// save expanded to temp file
+		if _, err = writer.Write([]byte(fmt.Sprintf("%s\n", srcContent))); err != nil {
+			return "", fmt.Errorf("unable to write to temp file: %s", err)
+		}
+		writer.Flush()
+	}
+	if err := tmpfile.Close(); err != nil {
+		return "", err
+	}
+	if err := srcFile.Close(); err != nil {
+		return "", err
+	}
+	return tmpfile.Name(), sc.Err()
 }

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -133,7 +133,10 @@ func (ob *OktetoBuilder) buildWithOkteto(ctx context.Context, buildOptions *type
 
 	// inject secrets to buildkit from temp folder
 	if err := replaceSecretsSourceEnvWithTempFile(afero.NewOsFs(), secretTempFolder, buildOptions); err != nil {
-		return err
+		return oktetoErrors.UserError{
+			E:    err,
+			Hint: "secret should have the format 'id=mysecret,src=/local/secret' where source exists as local file",
+		}
 	}
 
 	opt, err := getSolveOpt(buildOptions)
@@ -407,8 +410,6 @@ func GetVolumesToInclude(volumesToInclude []model.StackVolume) []model.StackVolu
 	return result
 }
 
-
-
 // replaceSecretsSourceEnvWithTempFile reads the content of the src of a secret and replaces the envs to mount into dockerfile
 func replaceSecretsSourceEnvWithTempFile(fs afero.Fs, secretTempFolder string, buildOptions *types.BuildOptions) error {
 	// for each secret at buildOptions extract the src
@@ -419,20 +420,20 @@ func replaceSecretsSourceEnvWithTempFile(fs afero.Fs, secretTempFolder string, b
 		csvReader := csv.NewReader(strings.NewReader(s))
 		fields, err := csvReader.Read()
 		if err != nil {
-			return err
+			return fmt.Errorf("error reading the csv secret, %w", err)
 		}
 
 		newFields := make([]string, len(fields))
 		for indx, field := range fields {
 			key, value, found := strings.Cut(field, "=")
 			if !found {
-				return fmt.Errorf("format error")
+				return fmt.Errorf("secret format error")
 			}
 
 			if key == "src" || key == "source" {
 				tempFileName, err := createTempFileWithExpandedEnvsAtSource(fs, value, secretTempFolder)
 				if err != nil {
-					return err
+					return fmt.Errorf("error creating the temp file with expanded values: %w", err)
 				}
 				value = tempFileName
 			}

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -396,17 +396,6 @@ func extractFromContextAndDockerfile(context, dockerfile, svcName string) string
 	return joinPath
 }
 
-// GetVolumesToInclude checks if the path exists, if it doesn't it skip it
-func GetVolumesToInclude(volumesToInclude []model.StackVolume) []model.StackVolume {
-	var result []model.StackVolume
-	for _, p := range volumesToInclude {
-		if _, err := os.Stat(p.LocalPath); err == nil {
-			result = append(result, p)
-		}
-	}
-	return result
-}
-
 // replaceSecretsSourceEnvWithTempFile reads the content of the src of a secret and replaces the envs to mount into dockerfile
 func replaceSecretsSourceEnvWithTempFile(fs afero.Fs, secretTempFolder string, buildOptions *types.BuildOptions) error {
 	// for each secret at buildOptions extract the src

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -133,10 +133,7 @@ func (ob *OktetoBuilder) buildWithOkteto(ctx context.Context, buildOptions *type
 
 	// inject secrets to buildkit from temp folder
 	if err := replaceSecretsSourceEnvWithTempFile(afero.NewOsFs(), secretTempFolder, buildOptions); err != nil {
-		return oktetoErrors.UserError{
-			E:    err,
-			Hint: "secret should have the format 'id=mysecret,src=/local/secret' where source exists as local file",
-		}
+		return fmt.Errorf("%w: secret should have the format 'id=mysecret,src=/local/secret'", err)
 	}
 
 	opt, err := getSolveOpt(buildOptions)

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -437,12 +437,10 @@ func parseTempSecrets(secretTempFolder string, buildOptions *types.BuildOptions)
 	// create a new file under tempFolder with the expanded content
 	// replace the src of the secret with the tempSrc
 	for indx, s := range buildOptions.Secrets {
-		splitSecret := strings.SplitN(s, "src=", 2)
-		if len(splitSecret) <= 0 {
-			return fmt.Errorf("unable to get secret src")
+		secretId, srcFileName, err := parseSecretFromString(s)
+		if err != nil {
+			return err
 		}
-		srcFileName := strings.TrimSpace(splitSecret[1])
-
 		// read source file
 		srcFile, err := os.Open(srcFileName)
 		if err != nil {
@@ -482,7 +480,7 @@ func parseTempSecrets(secretTempFolder string, buildOptions *types.BuildOptions)
 		}
 
 		// replace src
-		buildOptions.Secrets[indx] = fmt.Sprintf("%ssrc=%s", splitSecret[0], tmpfile.Name())
+		buildOptions.Secrets[indx] = fmt.Sprintf("id=%s,src=%s", secretId, tmpfile.Name())
 	}
 
 	return nil

--- a/pkg/cmd/build/build.go
+++ b/pkg/cmd/build/build.go
@@ -405,6 +405,31 @@ func GetVolumesToInclude(volumesToInclude []model.StackVolume) []model.StackVolu
 	return result
 }
 
+// parseSecretFromString returns id and src from the given secret string
+// format should be id=<name>,src=<source file>
+// if no id or src if found, error is returned
+func parseSecretFromString(s string) (string, string, error) {
+	_, afterId, found := strings.Cut(s, "id=")
+	if !found {
+		return "", "", fmt.Errorf("secret should have an id")
+	}
+	id, _, _ := strings.Cut(afterId, ",")
+	if id == "" {
+		return "", "", fmt.Errorf("secret id can not be empty")
+	}
+
+	_, afterSrc, found := strings.Cut(s, "src=")
+	if !found {
+		return "", "", fmt.Errorf("secret should have a source")
+	}
+	srcFileName, _, _ := strings.Cut(afterSrc, ",")
+	if srcFileName == "" {
+		return "", "", fmt.Errorf("secret source should not be empty")
+	}
+
+	return id, srcFileName, nil
+}
+
 // parseTempSecrets reads the content of the src of a secret and replaces the envs to mount into dockerfile
 func parseTempSecrets(secretTempFolder string, buildOptions *types.BuildOptions) error {
 	// for each secret at buildOptions extract the src

--- a/pkg/cmd/build/build_test.go
+++ b/pkg/cmd/build/build_test.go
@@ -624,3 +624,78 @@ func Test_parseTempSecrets(t *testing.T) {
 	require.Contains(t, string(b), envValue)
 
 }
+func Test_parseSecretFromString(t *testing.T) {
+	type args struct {
+		s string
+	}
+	tests := []struct {
+		name        string
+		args        args
+		expectedId  string
+		expectedSrc string
+		expectedErr bool
+	}{
+		{
+			name: "valid secret format",
+			args: args{
+				s: "id=mysecret,src=/local/path",
+			},
+			expectedId:  "mysecret",
+			expectedSrc: "/local/path",
+		},
+		{
+			name: "valid secret format 2",
+			args: args{
+				s: "src=/local/path,id=mysecret",
+			},
+			expectedId:  "mysecret",
+			expectedSrc: "/local/path",
+		},
+		{
+			name: "invalid secret format",
+			args: args{
+				s: "id=mysecret",
+			},
+			expectedId:  "",
+			expectedSrc: "",
+			expectedErr: true,
+		},
+		{
+			name: "invalid secret format 2",
+			args: args{
+				s: "src=/local/path",
+			},
+			expectedId:  "",
+			expectedSrc: "",
+			expectedErr: true,
+		},
+		{
+			name: "invalid secret format 3",
+			args: args{
+				s: "id=,src=/local/path",
+			},
+			expectedId:  "",
+			expectedSrc: "",
+			expectedErr: true,
+		},
+		{
+			name: "invalid secret format 4",
+			args: args{
+				s: "id=mysecret,src=",
+			},
+			expectedId:  "",
+			expectedSrc: "",
+			expectedErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			id, src, err := parseSecretFromString(tt.args.s)
+			require.Truef(t, tt.expectedErr == (err != nil), "expected error")
+
+			require.Equal(t, tt.expectedId, id)
+			require.Equal(t, tt.expectedSrc, src)
+
+		})
+	}
+}

--- a/pkg/cmd/build/build_test.go
+++ b/pkg/cmd/build/build_test.go
@@ -762,3 +762,49 @@ func Test_translateDockerErr(t *testing.T) {
 
 	}
 }
+
+func Test_setOutputMode(t *testing.T) {
+	tests := []struct {
+		name                     string
+		input                    string
+		envBuildkitProgressValue string
+		expected                 string
+	}{
+		{
+			name:     "not empty input",
+			input:    "outputmode",
+			expected: "outputmode",
+		},
+		{
+			name:     "empty input and empty env BUILDKIT_PROGRESS  - default output",
+			input:    "",
+			expected: "tty",
+		},
+		{
+			name:                     "empty input and unknown env BUILDKIT_PROGRESS  - default output",
+			input:                    "",
+			envBuildkitProgressValue: "unknown",
+			expected:                 "tty",
+		},
+		{
+			name:                     "empty input and plain env BUILDKIT_PROGRESS  - default output",
+			input:                    "",
+			envBuildkitProgressValue: "plain",
+			expected:                 "plain",
+		},
+		{
+			name:                     "empty input and json env BUILDKIT_PROGRESS  - default output",
+			input:                    "",
+			envBuildkitProgressValue: "json",
+			expected:                 "plain",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Setenv("BUILDKIT_PROGRESS", tt.envBuildkitProgressValue)
+			got := setOutputMode(tt.input)
+			require.Equal(t, tt.expected, got)
+		})
+	}
+}

--- a/pkg/cmd/build/build_test.go
+++ b/pkg/cmd/build/build_test.go
@@ -1,7 +1,6 @@
 package build
 
 import (
-	"bufio"
 	"bytes"
 	"fmt"
 	"log"
@@ -16,7 +15,7 @@ import (
 	"github.com/okteto/okteto/pkg/model"
 	"github.com/okteto/okteto/pkg/okteto"
 	"github.com/okteto/okteto/pkg/types"
-	"github.com/stretchr/testify/assert"
+	"github.com/spf13/afero"
 	"github.com/stretchr/testify/require"
 )
 
@@ -590,112 +589,132 @@ func removeFile(s string) error {
 	return nil
 }
 
-func Test_parseTempSecrets(t *testing.T) {
-	secretDir := t.TempDir()
-	tmpTestSecretFile, err := os.CreateTemp(secretDir, "")
+func Test_replaceSecretsSourceEnvWithTempFile(t *testing.T) {
+	t.Parallel()
+	fakeFs := afero.NewMemMapFs()
+	localSrcFile, err := afero.TempFile(fakeFs, t.TempDir(), "")
 	require.NoError(t, err)
 
-	writer := bufio.NewWriter(tmpTestSecretFile)
-	_, err = writer.Write([]byte(fmt.Sprintf("%s\n", "content for ${SECRET_ENV}")))
-	assert.NoError(t, err)
-	writer.Flush()
-
-	tempFolder := t.TempDir()
-	envValue := "secret env value"
-	t.Setenv("SECRET_ENV", envValue)
-
-	buildOpts := &types.BuildOptions{
-		Secrets: []string{
-			fmt.Sprintf("id=mysecret,src=%s", tmpTestSecretFile.Name()),
+	tests := []struct {
+		name                    string
+		fs                      afero.Fs
+		secretTempFolder        string
+		buildOptions            *types.BuildOptions
+		expectedErr             bool
+		expectedReplacedSecrets bool
+	}{
+		{
+			name:             "valid secret format",
+			fs:               fakeFs,
+			secretTempFolder: t.TempDir(),
+			buildOptions: &types.BuildOptions{
+				Secrets: []string{fmt.Sprintf("id=mysecret,src=%s", localSrcFile.Name())},
+			},
+			expectedErr:             false,
+			expectedReplacedSecrets: true,
+		},
+		{
+			name:             "valid secret format, reorder the fields",
+			fs:               fakeFs,
+			secretTempFolder: t.TempDir(),
+			buildOptions: &types.BuildOptions{
+				Secrets: []string{fmt.Sprintf("src=%s,id=mysecret", localSrcFile.Name())},
+			},
+			expectedErr:             false,
+			expectedReplacedSecrets: true,
+		},
+		{
+			name:             "valid secret format, only id",
+			fs:               fakeFs,
+			secretTempFolder: t.TempDir(),
+			buildOptions: &types.BuildOptions{
+				Secrets: []string{"id=mysecret"},
+			},
+			expectedErr: false,
+		},
+		{
+			name:             "valid secret format, only source",
+			fs:               fakeFs,
+			secretTempFolder: t.TempDir(),
+			buildOptions: &types.BuildOptions{
+				Secrets: []string{fmt.Sprintf("src=%s", localSrcFile.Name())},
+			},
+			expectedErr:             false,
+			expectedReplacedSecrets: true,
+		},
+		{
+			name:             "invalid secret, local file does not exist",
+			fs:               fakeFs,
+			secretTempFolder: t.TempDir(),
+			buildOptions: &types.BuildOptions{
+				Secrets: []string{"id=mysecret,src=/file/invalid"},
+			},
+			expectedErr:             true,
+			expectedReplacedSecrets: false,
 		},
 	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
 
-	require.NoError(t, parseTempSecrets(tempFolder, buildOpts))
+			initialSecrets := make([]string, len(tt.buildOptions.Secrets))
+			copy(initialSecrets, tt.buildOptions.Secrets)
+			err := replaceSecretsSourceEnvWithTempFile(tt.fs, tt.secretTempFolder, tt.buildOptions)
+			require.Truef(t, tt.expectedErr == (err != nil), "not expected error")
 
-	require.NoError(t, tmpTestSecretFile.Close())
+			if tt.expectedReplacedSecrets {
+				require.NotEqualValues(t, initialSecrets, tt.buildOptions.Secrets)
+			} else {
+				require.EqualValues(t, initialSecrets, tt.buildOptions.Secrets)
+			}
+		})
+	}
+}
 
-	require.NotContains(t, buildOpts.Secrets[0], tmpTestSecretFile.Name())
-	require.Contains(t, buildOpts.Secrets[0], "secret-")
+func Test_createTempFileWithExpandedEnvsAtSource(t *testing.T) {
+	fakeFs := afero.NewMemMapFs()
 
-	newSecretPath := strings.SplitN(buildOpts.Secrets[0], "id=mysecret,src=", 2)
-	b, err := os.ReadFile(newSecretPath[1])
+	localSrcFile, err := afero.TempFile(fakeFs, t.TempDir(), "")
 	require.NoError(t, err)
 
-	require.Contains(t, string(b), envValue)
+	_, err = localSrcFile.Write([]byte(`localEnv: ${ENV_IN_FILE}`))
+	require.NoError(t, err)
 
-}
-func Test_parseSecretFromString(t *testing.T) {
-	type args struct {
-		s string
-	}
 	tests := []struct {
 		name        string
-		args        args
-		expectedId  string
-		expectedSrc string
+		fakeFs      afero.Fs
+		sourceFile  string
+		envValue    string
 		expectedErr bool
 	}{
 		{
-			name: "valid secret format",
-			args: args{
-				s: "id=mysecret,src=/local/path",
-			},
-			expectedId:  "mysecret",
-			expectedSrc: "/local/path",
+			name:       "no error",
+			envValue:   "value of env",
+			sourceFile: localSrcFile.Name(),
+			fakeFs:     fakeFs,
 		},
 		{
-			name: "valid secret format 2",
-			args: args{
-				s: "src=/local/path,id=mysecret",
-			},
-			expectedId:  "mysecret",
-			expectedSrc: "/local/path",
-		},
-		{
-			name: "invalid secret format",
-			args: args{
-				s: "id=mysecret",
-			},
-			expectedId:  "",
-			expectedSrc: "",
-			expectedErr: true,
-		},
-		{
-			name: "invalid secret format 2",
-			args: args{
-				s: "src=/local/path",
-			},
-			expectedId:  "",
-			expectedSrc: "",
-			expectedErr: true,
-		},
-		{
-			name: "invalid secret format 3",
-			args: args{
-				s: "id=,src=/local/path",
-			},
-			expectedId:  "",
-			expectedSrc: "",
-			expectedErr: true,
-		},
-		{
-			name: "invalid secret format 4",
-			args: args{
-				s: "id=mysecret,src=",
-			},
-			expectedId:  "",
-			expectedSrc: "",
+			name:        "error - local file not exist",
+			sourceFile:  "file",
+			fakeFs:      fakeFs,
 			expectedErr: true,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			id, src, err := parseSecretFromString(tt.args.s)
-			require.Truef(t, tt.expectedErr == (err != nil), "expected error")
+			t.Setenv("ENV_IN_FILE", tt.envValue)
 
-			require.Equal(t, tt.expectedId, id)
-			require.Equal(t, tt.expectedSrc, src)
+			file, err := createTempFileWithExpandedEnvsAtSource(tt.fakeFs, tt.sourceFile, t.TempDir())
+			if tt.expectedErr {
+				require.Error(t, err)
+				return
+			}
 
+			f, err := afero.ReadFile(tt.fakeFs, file)
+			if !tt.expectedErr {
+				require.NoError(t, err)
+			}
+
+			require.Contains(t, string(f), "value of env")
 		})
 	}
 }

--- a/pkg/cmd/build/build_test.go
+++ b/pkg/cmd/build/build_test.go
@@ -652,6 +652,16 @@ func Test_replaceSecretsSourceEnvWithTempFile(t *testing.T) {
 			expectedErr:             true,
 			expectedReplacedSecrets: false,
 		},
+		{
+			name:             "invalid secret, no = found",
+			fs:               fakeFs,
+			secretTempFolder: t.TempDir(),
+			buildOptions: &types.BuildOptions{
+				Secrets: []string{"mysecret"},
+			},
+			expectedErr:             true,
+			expectedReplacedSecrets: false,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

--- a/pkg/cmd/build/build_test.go
+++ b/pkg/cmd/build/build_test.go
@@ -16,6 +16,7 @@ import (
 	"github.com/okteto/okteto/pkg/okteto"
 	"github.com/okteto/okteto/pkg/types"
 	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
@@ -726,5 +727,38 @@ func Test_createTempFileWithExpandedEnvsAtSource(t *testing.T) {
 
 			require.Contains(t, string(f), "value of env")
 		})
+	}
+}
+
+func Test_translateDockerErr(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       error
+		expectedErr error
+	}{
+		{
+			name:        "err is nil",
+			input:       nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "err is docker error",
+			input:       fmt.Errorf("failed to dial gRPC: cannot connect to the Docker daemon"),
+			expectedErr: errDockerDaemonConnection,
+		},
+		{
+			name:        "err is not docker error",
+			input:       assert.AnError,
+			expectedErr: assert.AnError,
+		},
+	}
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+			got := translateDockerErr(tt.input)
+			require.Equal(t, tt.expectedErr, got)
+		})
+
 	}
 }

